src/PhpSpreadsheet/Reader/Gnumeric.php

```diff
@@ -536,35 +566,54 @@ private function processColumnWidths(SimpleXMLElement $sheet, int $maxCol): void
     {
         if ((!$this->readDataOnly) && (isset($sheet->Cols))) {
             //    Column Widths
+            $defaultWidth = 0;
             $columnAttributes = $sheet->Cols->attributes();
-            $defaultWidth = $columnAttributes['DefaultSizePts'] / 5.4;
+            if ($columnAttributes !== null) {
+                $defaultWidth = $columnAttributes['DefaultSizePts'] / 5.4;
+            }
             $c = 0;
             foreach ($sheet->Cols->ColInfo as $columnOverride) {
                 $c = $this->processColumnLoop($c, $maxCol, $columnOverride, $defaultWidth);
             }
             while ($c <= $maxCol) {
-                $this->spreadsheet->getActiveSheet()->getColumnDimension(Coordinate::stringFromColumnIndex($c + 1))->setWidth($defaultWidth);
+                $this->setColumnWidth($c, $defaultWidth);
                 ++$c;
             }
         }
     }
 
+    private function setRowHeight(int $r, float $defaultHeight): void
```

Comment:
Can we use a more meaningful argument name than `$r`, e.g. `$rowNumber`? And expand abbreviated variable names like `$rowDim` to `$rowDimension`?

We should probably do the same for `$r` and `$c` in the main loop, but I know you inherited that, and it isn't really in scope for a PR that's intended to test the new approach to XML namespacing, so that might be a later PR just to make variable names more explicit

Reply:
I have renamed 'c' and 'r' throughout to 'whichColumn' and 'whichRow', in order to avoid conflicting with existing 'row' and 'column' variables. I suppose rowNumber and columnLetter might have been better choices. Also changed rowDim->rowDimension and colDum->columnDimension.